Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -499,6 +499,47 @@ WarpXParticleContainer::DepositCurrent (WarpXParIter& pti,
 #endif
 }
 
+void
+WarpXParticleContainer::DepositCurrent (
+    amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& J,
+    const amrex::Real dt, const amrex::Real relative_t)
+{
+    // Loop over the refinement levels
+    int const finest_level = J.size() - 1;
+    for (int lev = 0; lev <= finest_level; ++lev)
+    {
+        // Loop over particle tiles and deposit current on each level
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+        {
+        int thread_num = omp_get_thread_num();
+#else
+        int thread_num = 0;
+#endif
+        for (WarpXParIter pti(*this, lev); pti.isValid(); ++pti)
+        {
+            const long np = pti.numParticles();
+            auto & wp = pti.GetAttribs(PIdx::w);
+            auto & uxp = pti.GetAttribs(PIdx::ux);
+            auto & uyp = pti.GetAttribs(PIdx::uy);
+            auto & uzp = pti.GetAttribs(PIdx::uz);
```

Comment:
We can even be more explicit since we only read:
```suggestion
            auto const & wp = pti.GetAttribs(PIdx::w);
            auto const & uxp = pti.GetAttribs(PIdx::ux);
            auto const & uyp = pti.GetAttribs(PIdx::uy);
            auto const & uzp = pti.GetAttribs(PIdx::uz);
```

Reply:
Done in https://github.com/ECP-WarpX/WarpX/pull/1828/commits/971665564e46b1f24df0308c886ccbd8018fab90.